### PR TITLE
[tree] Add missing `override` specifiers in tree/tree*

### PR DESCRIPTION
**Pull request by guitargeek**
The first commit adds the missing `override` keywords in the tree/tree* sources as suggested by clang-tidy.

The second commits makes all the necessary manual changes to not get inconsistent override warnings after the `override` keywords have been added.

### CODE REVIEW

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
I'll close this for now because I have no time to figure out why the tests are still not passing.

**Comment by guitargeek** [comment]
Let's see what happens here, now that the `roottest` PR is merged:
https://github.com/root-project/roottest/pull/960

File: tree/treeplayer/inc/TFormLeafInfo.h [hunk 1]
```diff
@@ -44,6 +44,26 @@
    template<typename T> T  ReadValueImpl(char *where, Int_t instance = 0)  // no semicolon
 
 
+// declare the extra versions of GetValue() plus templated implementation
+#define DECLARE_GETVAL_OVERRIDE \
+   Double_t  GetValue(TLeaf *leaf, Int_t instance = 0) override               \
+       { return GetValueImpl<Double_t>(leaf, instance); }                     \
```
Comment (pcanal) [inline tree/treeplayer/inc/TFormLeafInfo.h:50]: What about using only one (well 2) macros:
```suggestion
#define DECLARE_GETVAL(VIRTUAL, OVERRIDE) \
   VIRTUAL Double_t  GetValue(TLeaf *leaf, Int_t instance = 0) OVERRIDE               \
       { return GetValueImpl<Double_t>(leaf, instance); }                                              \
```
and usage:
```
DECLARE_GETVAL(virtual,)
// or
DECLARE_GETVAL( , override)
```

Comment (guitargeek) [reply tree/treeplayer/inc/TFormLeafInfo.h]: I agree, it's a nice way to avoid the code duplication! I have updated the PR.

**Review by pcanal — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**